syntax = "proto3";
package project_common;

// Data type enumeration
enum DataType {
  DATA_TYPE_UNSPECIFIED = 0;  // Default, should not be used
  DATA_TYPE_FILE = 1;         // File data
  DATA_TYPE_PYTHON_PICKLE = 2; // Python pickled object
}

message StatusResponse {
  bool success = 1;
  string message = 2;
}

message DataIdentifier {
  string id = 1;
}

message FileIdentifier {
  string id = 1;
}

// Metadata about data before streaming
message DataMetadata {
  bool found = 1;                 // True if data_id was found
  DataType data_type = 2;         // Type of the data being streamed
  int64 total_size_bytes = 3;     // Total size of the data to be streamed
  string error_message = 4;       // Message if found is false or an error occurred
}

// Chunk of data for streaming
message DataChunk {
  bytes chunk_data = 1;           // Raw data chunk
  bool is_last_chunk = 2;         // True if this is the final chunk
}